engine/flink/components-utils/src/main/scala/pl/touk/nussknacker/engine/flink/util/source/BlockingQueueSource.scala

```diff
@@ -38,44 +38,17 @@ class BlockingQueueSource[T](returnType: TypingResult, extractTimestampFun: T =>
 
   private val contextInitializer = new BasicContextInitializer[T](returnType)
 
-  private def flinkSourceFunction: SourceFunction[T] = {
-    // extracted for serialization purpose
-    val copyOfExtractTimestampFun = extractTimestampFun
-    val copyOfId                  = id
-    new SourceFunction[T] {
-
-      @volatile private var isRunning = true
-
-      private val id = copyOfId
-
-      override def run(ctx: SourceFunction.SourceContext[T]): Unit = {
-        val queue = BlockingQueueSource.getForId[T](id)
-        while (isRunning) {
-          Option(queue.poll(100, TimeUnit.MILLISECONDS)).foreach {
-            case Some(element) =>
-              val timestamp = copyOfExtractTimestampFun(element)
-              ctx.collectWithTimestamp(element, timestamp)
-
-              val watermark = new Watermark(timestamp - maxOutOfOrderness.toMillis)
-              ctx.emitWatermark(watermark)
-            case None =>
-              isRunning = false
-          }
-        }
-      }
-
-      override def cancel(): Unit = isRunning = false
-
-    }
-  }
-
   override def contextStream(
       env: StreamExecutionEnvironment,
       flinkNodeContext: FlinkCustomNodeContext
   ): DataStream[Context] = {
     env
-      .addSource(flinkSourceFunction, TypeInformationDetection.instance.forType[T](returnType))
-      .name(s"${flinkNodeContext.metaData.name}-${flinkNodeContext.nodeId}-source")
+      .fromSource(
+        new BlockingQueueFlinkSource[T](id, extractTimestampFun, maxOutOfOrderness),
+        WatermarkStrategy.noWatermarks(),
+        s"${flinkNodeContext.metaData.name}-${flinkNodeContext.nodeId}-source",
+        TypeInformationDetection.instance.forType[T](returnType)
+      )
```

Comment:
Shouldn't we also assign a UID, for the sake of consistency, and for people who may be copying this code to make their own source? We already do this for `CollectionSource`.